evm/src/cpu/stack_bounds.rs

```diff
@@ -0,0 +1,132 @@
+//! Checks for stack underflow and overflow.
+
+use plonky2::field::extension::Extendable;
+use plonky2::field::packed::PackedField;
+use plonky2::field::types::Field;
+use plonky2::hash::hash_types::RichField;
+use plonky2::iop::ext_target::ExtensionTarget;
+
+use crate::constraint_consumer::{ConstraintConsumer, RecursiveConstraintConsumer};
+use crate::cpu::columns::{CpuColumnsView, COL_MAP};
+
+const MAX_STACK_SIZE: u64 = 1024;
+
+const DECREMENTING_FLAGS: [usize; 1] = [COL_MAP.is_pop];
```

Comment:
Let's add a comment noting why we don't need other decrementing instructions to check underflow.